content/francais/python-turtle/activity-4.md

```diff
@@ -0,0 +1,77 @@
+---
+title: "Activité 4 - Simplifier le code avec des fonctions"
+date: 2024-09-17T13:24:17-07:00
+weight: 6
+draft: false
+---
+
+Nous avons dessiné un hexagone. Mais ce n'est qu'un des nombreux hexagones qu'Alex doit construire pour sa ruche. Encore une fois, il semble que nous devrions répéter le code plusieurs fois pour dessiner plusieurs hexagones. Heureusement, nous n'en avons pas besoin ! Les fonctions à la rescousse !
+
+{{% notice tip %}}
+
+## Fonctions
+
+Une fonction est un moyen de regrouper des lignes de code pour faire quelque chose. Par exemple, `turtle.forward(50)` déplace la tortue de 50 pas en avant, et `turtle.left(120)` tourne la tortue de 120 degrés à gauche. Que faire si nous voulons toujours effectuer ces deux actions ensemble ? Nous pouvons les mettre dans une fonction appelée `draw_line()`, de sorte que chaque fois que cette fonction est appelée, ces deux actions se produisent simultanément. Pensez-y comme une recette : nous regroupons toutes les instructions ensemble, et le nom de la recette est la nourriture que nous préparons.
+
+{{% /notice %}}
+
+Voici un exemple de **définition de fonction**, et comment nous l'utiliserions :
+
+``` python
+def draw_line():
+  turtle.forward(50)
+  turtle.left(120)
+```
+
+La première ligne est appelée l'**en-tête** de définition de fonction. Le mot-clé `def` indique à l'ordinateur que nous définissons une nouvelle fonction. Ensuite, nous donnons un nom à la fonction, dans ce cas `draw_line`. Enfin, n'oubliez pas le `():` à la fin de la ligne, qui indique à l'ordinateur que nous commençons le corps de la fonction.
+
+{{% notice tip %}}
+
+## Matériel supplémentaire - paramètres
+
+Dans les parenthèses `()` nous définirions normalement des **paramètres** pour la fonction. Les paramètres sont des entrées que nous pouvons donner à la fonction, que la fonction peut utiliser pour produire sa sortie. Par exemple, nous devons avoir deux nombres pour les additionner et produire un résultat - les deux nombres que nous utilisons peuvent être considérés comme des **paramètres**. Aujourd'hui, nous n'utiliserons pas de paramètres, et nous laisserons comme `()`.
+
+{{% /notice %}}
+
+Semblable à une boucle for, tout ce que nous souhaitons définir dans la fonction doit être précédé d'une tabulation.
```

Comment:
for-loop was translated literally... not sure if this is the correct way of handling this

Reply:
This is correct if we want to say it in French

Reply:
Keeping `for loop` may also be ok